    void \u02cf(int, int, byte[]) throws java.io.IOException
    {
        java.lang.Throwable $r2;
        byte[] $r1;
        java.lang.NullPointerException r3;
        util.h.xy.cy.mf r0;
        int $i0, $i1, $i2, $i3;
        boolean $z0;

        r0 := @this: util.h.xy.cy.mf;

        $i0 := @parameter0: int;

        $i1 := @parameter1: int;

        $r1 := @parameter2: byte[];

        $i2 = <util.h.xy.cy.mf: int \u02bb>;

        $i2 = $i2 + 105;

        $i3 = $i2 % 128;

        <util.h.xy.cy.mf: int \u02bc> = $i3;

        virtualinvoke r0.<util.h.xy.cy.mf: void \u141d(int,int)>($i0, $i1);

        $i0 = lengthof $r1;

        virtualinvoke r0.<util.h.xy.cy.mf: void \u02ca(int)>($i0);

        virtualinvoke r0.<util.h.xy.cy.mf: void \u02ce(byte[])>($r1);

        $i0 = <util.h.xy.cy.mf: int \u02bc>;

        $i0 = $i0 + 93;

        $i1 = $i0 % 128;

        <util.h.xy.cy.mf: int \u02bb> = $i1;

        $z0 = 0;

        $i0 = $i0 % 2;

        if $i0 == 0 goto label1;

        $z0 = 1;

     label1:
        if $z0 == 0 goto label4;

     label2:
        r3 = new java.lang.NullPointerException;

        specialinvoke r3.<java.lang.NullPointerException: void <init>(java.lang.String)>("This statement would have triggered an Exception: specialinvoke $u3#16.<java.lang.Object: int hashCode()>()");

        throw r3;

     label3:
        $r2 := @caughtexception;

        throw $r2;

     label4:
        return;

        catch java.lang.Throwable from label2 to label3 with label3;
    }
